src/hooks/useAzureMapLayer.test.tsx

```diff
@@ -0,0 +1,173 @@
+import { render } from '@testing-library/react'
+
+import atlas, { layer, source } from 'azure-maps-control'
+import { constructLayer, useAzureMapLayer } from './useAzureMapLayer'
+import React from 'react'
+import { AzureMapLayerContext, AzureMapLayerProvider } from '../contexts/AzureMapLayerContext'
+import { AzureMapDataSourceContext } from '../contexts/AzureMapDataSourceContext'
+import { IAzureMapDataSourceContextState, IAzureMapLayerContextState } from '../types'
+
+const datasourceRef = {} as source.DataSource
+
+const renderWithDataSourceContext = (node: React.ReactNode) => {
```

Comment:
we should take this and put into helper test methods, the same for renderWithMapRef